### PR TITLE
Added AdditionalUserInfo

### PR DESCRIPTION
**Pull request by scottcrossen**

### CODE REVIEW

File: packages/auth-ts/src/api/authentication.ts [hunk 1]
```diff
@@ -215,7 +216,6 @@ export interface SignInWithIdpRequest {
 }
 
 export interface SignInWithIdpResponse extends IdTokenResponse {
-  providerId?: string;
```
Comment (samhorlbeck) [inline packages/auth-ts/src/api/authentication.ts]: why did this get removed?

Comment (samhorlbeck) [reply packages/auth-ts/src/api/authentication.ts]: never mind

File: packages/auth-ts/src/model/additional_user_info.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    additionalInfo.set(userCredential, additionalUserInfo);
+  }
+}
+
+const additionalInfo = new WeakMap<UserCredential, AdditionalUserInfo>();
+
+function additionalUserInfoFromIdTokenResponse(
+  idTokenResponse: IdTokenResponse
+): AdditionalUserInfo | null {
+  const providerId = idTokenResponse.providerId;
```
Comment (samhorlbeck) [inline packages/auth-ts/src/model/additional_user_info.ts]: const {providerId} = idTokenResponse

File: packages/auth-ts/src/model/additional_user_info.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  const profile =
+    typeof idTokenResponse.rawUserInfo === 'string'
+      ? JSON.parse(idTokenResponse.rawUserInfo)
+      : {};
+  const isNewUser =
+    !!idTokenResponse.isNewUser ||
+    idTokenResponse.kind === 'identitytoolkit#SignupNewUserResponse';
+  if (!providerId && idTokenResponse.idToken !== undefined) {
```
Comment (samhorlbeck) [inline packages/auth-ts/src/model/additional_user_info.ts]: (!providerId && idTokenResponse) {   -- no need to check strict inequality with undefined

File: packages/auth-ts/src/model/additional_user_info.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    idTokenResponse
+  );
+  if (!!additionalUserInfo) {
+    additionalInfo.set(userCredential, additionalUserInfo);
+  }
+}
+
+const additionalInfo = new WeakMap<UserCredential, AdditionalUserInfo>();
+
+function additionalUserInfoFromIdTokenResponse(
```
Comment (samhorlbeck) [inline packages/auth-ts/src/model/additional_user_info.ts]: This shouldn't be in model. I'd put it in core/account_management

File: packages/auth-ts/src/core/strategies/auth_credential.ts [hunk 5]
```diff
@@ -122,7 +125,7 @@ export function authCredentialFromTokenResponse(
         pendingToken,
         idToken: oauthIdToken,
         accessToken,
-        signInMethod: providerId as SignInMethod
+        signInMethod: (providerId as unknown) as SignInMethod
```
Comment (samhorlbeck) [inline packages/auth-ts/src/core/strategies/auth_credential.ts]: This is funky

Comment (scottcrossen) [reply packages/auth-ts/src/core/strategies/auth_credential.ts]: It wont compile without it: "There is no overlap between these two types"

Comment (samhorlbeck) [reply packages/auth-ts/src/core/strategies/auth_credential.ts]: Try casting to string instead of unknown then

**Comment by scottcrossen** [comment]
Build locally. The issue is with another package.

**Pull request merged**